guide/ops/troubleshooting/memory-usage.md

```diff
@@ -0,0 +1,138 @@
+Many common causes of leaks, including as runaway tasks and cyclic dependent configuration,
+will show their own log errors prior to the memory error.
+
+You should also note the task counts in the `brooklyn gc` messages described above,
+and if there are an exceptional number of tasks or tasks are not clearing,
+other log messages will describe what is happening, and the in-product task
+view can indicate issues. 
+
+Sometimes slow leaks can occur if blueprints do not clean up entities or locations.
+These can be diagnosed by noting the number of files written to the persistence location,
+if persistence is being used.  Deploying then destroying a blueprint should not leave
+anything behind in the persistence directory.
+
+Where problems have been encountered in the past, we have resolved them and/or
+worked to improve logging and early identification.
+Please report any issues so that we can improve this further.
+In many cases we can also give advice on what other log `grep` patterns can be useful.
+
+
+### Standard Java Techniques
+
+Useful standard Java techniques for tracking memory leaks include:
+
+* `jstack <pid>` to see what tasks are running
+* `jmap -histo:live <pid>` to see what objects are using memory (see below)
+* Memory profilers such as VisualVM or Eclipse MAT, either connected to a running system or
+  against a heap dump generated on an OOME
+
+More information is available on [the Oracle Java web site](https://docs.oracle.com/javase/7/docs/webnotes/tsg/TSG-VM/html/memleaks.html).
+
+Note that some of the above techniques will often include soft and weak references that are irrelevant
+to the problem (and will be cleared on an OOME). Objects that may be cached in that way include:
+
+* `BasicConfigKey` (used for the web server and many blueprints)
+* `DslComponent` and `*Task` (used for Brooklyn activities and dependent configuration)
+* `jclouds` items including `ImageImpl` (to cache data on cloud service providers)
+
+On the other hand any of the above may also indicate a leak.
+Taking snapshots after a `forceClearSoftReferences()` (above) invocation and comparing those
+is one technique to filter out noise.  Another is to wait until there is an OOME
+and look just after, because that will clear all non-essential data from memory.
+(The `forceClearSoftReferences()` actually works by triggering an OOME, in as safe 
+a way as possible.)
+
+If leaked items are found, a profiler will normally let you see their content
+and walk backwards along their references to find out why they are being retained.
+
+
+### Summary of Techniques
+
+The following sequence of techniques is a common approach to investigating and fixing memory issues:
+
+* Note the log lines about `brooklyn gc`, including memory and tasks
+* Do not assume high memory usage alone is an error, as soft reference caches are deliberate; 
+  use `forceClearSoftReferences()` to clear these
```

Comment:
we say

| This call can be disruptive to the system however so use with care.

and

| Many JVM authorities discourage interfering with its garbage collector, however,
| so use with care and study the particular JVM you are using.

this felt sufficient but if you still want i'll add more caveats.

note the code fails because it tries to allocate a huge block of memory:  any parallel code trying to allocate a non-huge amount of memory should be unaffected.